packit/schema.py

```diff
@@ -1,82 +1,254 @@
+    Can be passed additional message via additional_message parameter.
+
+    :param str additional_message: additional warning message to be displayed
+    """
+
+    def _serialize(self, value: typing.Any, attr: str, obj: typing.Any, **kwargs):
+        raise NotImplementedError
+
+    def _deserialize(
+        self,
+        value: typing.Any,
+        attr: typing.Optional[str],
+        data: typing.Optional[typing.Mapping[str, typing.Any]],
+        **kwargs,
+    ):
+        logger.warning(f"{self.name} is no longer being processed.")
+        additional_message = self.metadata.get("additional_message", None)
+        if additional_message:
+            logger.warning(f"{additional_message}")
+
+
+class SyncFilesConfigSchema(Schema):
+    """
+    Schema for processing SyncFilesConfig config data.
+    """
+
+    files_to_sync = fields.List(FilesToSyncField(allow_none=False), allow_none=True)
+
+    @post_load
+    def make_instance(self, data, **kwargs):
+        return SyncFilesConfig(**data)
+
+    @pre_load
+    def list_to_dict(self, data, **kwargs):
+        """
+        If files are provided as list[str] not as schema dict, input data format is modified
+        to follow schema layout
+
+        ex.
+        [f1,f2,..] -> {"files_to_sync": [f1, f2, f3, ...]}
+        """
+
+        if isinstance(data, list):
+            return {"files_to_sync": data}
+        else:
+            return data
+
+
+class JobConfigSchema(Schema):
+    """
+    Schema for processing JobConfig config data.
+    """
+
+    class Meta:
+        unknown = RAISE
+
+    job = EnumField(JobType, required=True)
+    trigger = EnumField(JobTriggerType, required=True)
+    metadata = fields.Dict(missing={})
+
+    @post_load
+    def make_instance(self, data, **kwargs):
+        return JobConfig(**data)
+
+
+class PackageConfigSchema(Schema):
+    """
+    Schema for processing PackageConfig config data.
+    """
+
+    class Meta:
+        unknown = RAISE
+
+    config_file_path = fields.String()
+    specfile_path = fields.String(required=True)
+    downstream_package_name = fields.String()
+    upstream_project_url = fields.String(missing=None)
+    upstream_package_name = fields.String()
+    upstream_ref = fields.String()
+    upstream_tag_template = fields.String()
+    dist_git_url = NotProcessedField(
+        additional_message="it is generated from dist_git_base_url and downstream_package_name"
+    )
+    dist_git_base_url = fields.String()
+    dist_git_namespace = fields.String()
+    create_tarball_command = fields.List(fields.String())
+    current_version_command = fields.List(fields.String())
+    allowed_gpg_keys = fields.List(fields.String())
+    spec_source_id = fields.Method(deserialize="spec_source_id_fm")
+    synced_files = fields.Nested(SyncFilesConfigSchema)
+    jobs = fields.Nested(JobConfigSchema, many=True, missing=default_jobs_dict)
+    actions = fields.Dict(
+        keys=EnumField(ActionName, by_value=True)
```

Comment:
I meant this was causing test failures as the key in the Config was saved just as an enum value e.q. not ActionName.pre_sync, just pre-sync